HOWTORELEASE.rst

```diff
@@ -27,10 +27,6 @@ Note: this assumes you have already registered on pypi.
        devpi list pytest
 
    or look at failures with "devpi list -f pytest".
-   There will be some failed environments like e.g. the py33-trial 
```

Comment:
This note no longer holds True since tox supports skipping certain environments based on the platform. From AppVeyor:

```
========================== 8 passed in 7.47 seconds ===========================
___________________________________ summary ___________________________________ 
  flakes: commands succeeded
  py26: commands succeeded
  py27: commands succeeded
  py33: commands succeeded
  py34: commands succeeded
  py35: commands succeeded
ERROR:   pypy: InterpreterNotFound: pypy
SKIPPED:  py27-pexpect: platform mismatch
  py27-xdist: commands succeeded
  py27-trial: commands succeeded
SKIPPED:  py35-pexpect: platform mismatch
  py35-xdist: commands succeeded
SKIPPED:  py35-trial: platform mismatch
  py27-nobyte: commands succeeded
  doctesting: commands succeeded
SKIPPED:  py27-cxfreeze: platform mismatch
```